tests/unit/orchestrators/test_utils.py

```diff
@@ -33,7 +34,7 @@ def test_get_cache_status_raises_no_error_when_none_passed():
         pytest.fail("`get_cache_status()` raised an `AttributeError`")
 
 
-def test_get_cache_status_works_when_running_pipeline_twice(clean_repo):
+def test_get_cache_status_works_when_running_pipeline_twice():
```

Comment:
Sure. I created a new, `autouse` module-level fixture called `base_profile` that creates a separate profile for every unit test module. This means that `clean_repo` is no longer required, because that guarantees separation of stacks at module level. The only tests that still need a `clean_repo` (meaning a completely new global configuration and set of profiles after my PR) are the repository and global configuration tests. Even the examples benefit from the same profile separation. 

The bonus is that profiles are now also actively being tested as part of the test setup itself.

Reply:
Ah that clears at least one of my concerns, I somehow mistakenly remembered that the `base_profile` was a session-scoped fixture similar to `base_repo`. I'm guessing we could still run into trouble if the same testing module ran a pipeline with the same name and step names, in which case the caching results might be wrong depending on the order of the tests? Probably not a real concern and seems to not happen in any tests, just something to keep in mind for future tests. I see the risk especially in modules that use our fixtures to get pipelines/steps, in which case the overlap might happen more frequently